From e5c30c588891eb8c36ca37c6fdf905bcfdf394f3 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Wed, 27 Jan 2021 20:41:51 +0100
Subject: [PATCH 1/1] cmd: load, emit error message for invalid block device

The load command should not silently return to the console prompt if an
invalid block device is specified and no file is loaded.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 fs/fs.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/fs/fs.c b/fs/fs.c
index 68a15553cc..0c8f577a79 100644
--- a/fs/fs.c
+++ b/fs/fs.c
@@ -710,8 +710,10 @@ int do_load(struct cmd_tbl *cmdtp, int flag, int argc, char *const argv[],
 	if (argc > 7)
 		return CMD_RET_USAGE;
 
-	if (fs_set_blk_dev(argv[1], (argc >= 3) ? argv[2] : NULL, fstype))
+	if (fs_set_blk_dev(argv[1], (argc >= 3) ? argv[2] : NULL, fstype)) {
+		log_err("Can't set block device\n");
 		return 1;
+	}
 
 	if (argc >= 4) {
 		addr = simple_strtoul(argv[3], &ep, 16);
-- 
2.29.2

